src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -1086,13 +1094,11 @@ void RealmCoordinator::run_async_notifiers()
         // attaching A, then advancing to version 5 (letting A gather changes
         // from 2-5). We then attach C and advance to 7, then attach B and advance
         // to the latest.
-        std::sort(new_notifiers.begin(), new_notifiers.end(), compare_notifier_versions);
-        new_notifier_transaction = m_db->start_read(new_notifiers.front()->version());
+        new_notifier_transaction = m_db->start_read(version);
```

Comment:
This feels very suspicious because anything which actually uses this transaction will get incorrect data. It looks like it might be okay because too-new schema information can only result in a bit of extra work being done and discovering that a collection will eventually be deleted "too early" is actually a perf optimization that gives correct results, but it's hard to be confident that it's correct.

Reply:
add_required_change_info() (recursively) checks which tables the query's table links to determine which tables we need to track changes on. This could probably be reworked to instead use the Schema, which would make it so that we don't need a read transaction at all here.